website/src/controllers/api-v3/auth.js

```diff
@@ -124,7 +123,8 @@ api.registerLocal = {
 
     if (fbUser) {
       if (!fbUser.auth.facebook.id) throw new NotAuthorized(res.t('onlySocialAttachLocal'));
-      fbUser.auth.local = newUser;
+      fbUser.auth.local = newUser.auth.local;
+      fbUser.preferences = newUser.preferences;
```

Comment:
I see, in this case I think you can just do `fbUser.auth.local = newUser.auth.local ` and skip assigning the preferences, that object only contains the language preference and it's not important for returning users also we would risk overwriting any information in fbUser.preferences